vxquery-core/src/main/javacc/xquery-grammar.jj

```diff
@@ -1937,6 +1938,47 @@ ASTNode FunctionCall()  :
     }
 }
 
+ASTNode JsonConstructor()  :
+{
+    ASTNode result;
+}
+{
+    result = ArrayConstructor()
+    {
+		return result;
+	}
+}
+
+ASTNode ArrayConstructor()  :
+{
+	Token start, end;
+	List<ASTNode> items;
+}
+{
+	start="[" items=ArrayItemList() end="]" {
+		ArrayConstructorNode array = new ArrayConstructorNode(createSourceLocation(start));
+		array.setItems(items);
+		return array;
+	}
+}
+
+List<ASTNode> ArrayItemList()  :
```

Comment:
If we just reuse the expression's ability to capture a sequence, we could make the production of the ArrayConstructor `"[" expr = Expr() "]"` and avoid this ArrayItemList.